heft-plugins/heft-isolated-typescript-transpile-plugin/src/SwcIsolatedTranspilePlugin.ts

```diff
@@ -0,0 +1,384 @@
+  }: IEmitKind): IOptionsBufferByExtension {
+    const format: ModuleConfig['type'] | undefined =
+      formatOverride !== undefined ? TSC_TO_SWC_MODULE_MAP[formatOverride] : undefined;
+    if (format === undefined) {
+      throw new Error(`Unsupported Module Kind: ${formatOverride && ts.ModuleKind[formatOverride]} for swc`);
+    }
+
+    logger.terminal.writeVerboseLine(`Transpiling to format: ${format}`);
+
+    const target: JscTarget | undefined =
+      targetOverride !== undefined ? TSC_TO_SWC_TARGET_MAP[targetOverride] : undefined;
+    if (target === undefined) {
+      throw new Error(`Unsupported Target: ${target && ts.ScriptTarget[target]} for swc`);
+    }
+
+    logger.terminal.writeVerboseLine(`Transpiling to target: ${target}`);
+
+    const moduleConfig: ModuleConfig = {
+      type: format,
+      noInterop: tsConfigOptions.esModuleInterop === false
+    };
+
+    const parser: ParserConfig = {
+      syntax: 'typescript',
+      decorators: experimentalDecorators,
+      dynamicImport: true,
+      tsx: false
+    };
+
+    // https://github.com/swc-project/swc-node/blob/e6cd8b83d1ce76a0abf770f52425704e5d2872c6/packages/register/read-default-tsconfig.ts#L131C7-L139C20
+    const react: Partial<ReactConfig> | undefined =
+      tsConfigOptions.jsxFactory ??
+      tsConfigOptions.jsxFragmentFactory ??
+      tsConfigOptions.jsx ??
+      tsConfigOptions.jsxImportSource
+        ? {
+            pragma: tsConfigOptions.jsxFactory,
+            pragmaFrag: tsConfigOptions.jsxFragmentFactory,
+            importSource: tsConfigOptions.jsxImportSource ?? 'react',
+            runtime: (tsConfigOptions.jsx ?? 0) >= ts.JsxEmit.ReactJSX ? 'automatic' : 'classic',
+            useBuiltins: true
+          }
+        : undefined;
+
+    let options: SwcOptions = {
+      cwd: buildFolderPath,
+      root: srcDir,
+      rootMode: 'root',
+      configFile: false,
+      swcrc: false,
+      minify: false,
+
+      inputSourceMap: false,
+      sourceRoot,
+      isModule: true,
+
+      module: moduleConfig,
+      jsc: {
+        target,
+        externalHelpers: tsConfigOptions.importHelpers,
+        parser,
+        transform: {
+          legacyDecorator: experimentalDecorators,
+          react,
+          useDefineForClassFields,
+          // This property is not included in the types, but is what makes swc-jest work
+          // @ts-ignore
+          hidden: {
+            jest: format === 'commonjs'
+          }
+        }
+      }
+    };
+
+    if (getSwcOptionsHook.isUsed()) {
+      options = getSwcOptionsHook.call(options, formatOverride, targetOverride);
+    }
+
+    logger.terminal.writeVerboseLine(`Transpile options: ${JSON.stringify(options, undefined, 2)}}`);
+    logger.terminal.writeDebugLine(`Transpile options: ${options}`);
+
+    const tsOptions: string = JSON.stringify(options);
+    const tsOptionsBuffer: Buffer = Buffer.from(new SharedArrayBuffer(Buffer.byteLength(tsOptions, 'utf8')));
```

Comment:
Using SharedArrayBuffer here is irrelevant since this will be passed to a child_process over a socket. There's a crash when combining `worker_threads` with `@swc/core`.